types/src/v0/impls/header.rs

```diff
@@ -807,8 +797,10 @@ impl BlockHeader<SeqTypes> for Header {
         let block_merkle_tree_root = block_merkle_tree.commitment();
         let fee_merkle_tree_root = fee_merkle_tree.commitment();
 
+        // The Header is versioned, so we check if there are any upgrades.
+        // If no upgrades are specified, we use the default V1 Header (version 0.1).
+        // If an upgrade is specified, we use the version from the latest upgrade.
         let upgrades = instance_state.upgrades.last_key_value();
```

Comment:
Actually shouldn't the genesis header always be 0.1? Because it would be from before any upgrades had taken effect

Reply:
I think we should use the current_version inside the NodeState.
so if the base version is 0.2, we build V2 Header